### PR TITLE
chore: sync v3 with main branch

### PR DESCRIPTION
**Pull request by tswast**
"do not merge" label is just a reminder to use a merge commit

### CODE REVIEW

File: tests/unit/test_table.py [hunk 1]
```diff
@@ -2205,6 +2205,25 @@ def test_to_arrow_iterable(self):
             [[{"name": "Bepples Phlyntstone", "age": 0}, {"name": "Dino", "age": 4}]],
         )
 
+    @mock.patch("google.cloud.bigquery.table.pyarrow", new=None)
+    def test_to_arrow_iterable_error_if_pyarrow_is_none(self):
```
Comment (tswast) [inline tests/unit/test_table.py:2209]: Oops. I think we can delete this now that pyarrow is a required dependency

**Comment by tswast** [comment]
Note: the diff is a bit smaller than expected because https://github.com/googleapis/python-bigquery/pull/1081 pulled some changes in from `main` already

**Review by shollyman — APPROVED** [review]
LGTM, with the caveat that I have no real context on what's in/out of scope for the merge as I've not been following the v3 branch closely.

**Comment by plamut** [comment]
> LGTM, with the caveat that I have no real context on what's in/out of scope for the merge as I've not been following the v3 branch closely.

Everything from the `main` branch should go into `v3`, but of course adjusted for the changes in that branch (e.g. some dependencies not optional anymore, better type hints, etc.).

**Review by plamut — APPROVED** [review]
LGTM.

The failures seem transient, a bunch of 500 errors.

**Comment by plamut** [comment]
Hm... the copy table test still consistently fails across all test runs.

**Comment by tswast** [comment]
> Hm... the copy table test still consistently fails across all test runs.

Seth filed internal issue 210907595 to get this fixed.

**Pull request merged**